File: seungmi/week10/입국심사.py
def solution(n, times):
    answer = 0
    
    left = min(times)
    right = max(times)*n
    while left <= right:
        mid = (left+ right) // 2
        checked = 0
        for time in times:
            checked += mid // time

            if checked >= n:
                break
        
        # 심사한 사람의 수가 심사 받아야할 사람의 수(n)보다 거나 같은 경우
        if checked >= n:
            answer = mid
            right = mid - 1
        # 심사한 사람의 수가 심사 받아야할 사람의 수(n)보다 적은 경우
        elif checked < n:
            left = mid + 1
            
    return answer

File: sujeong/week10/입국심사.swift
import Foundation

func solution(_ n:Int, _ times:[Int]) -> Int64 {
    guard let maxTime:Int = times.max() else { return 0 }
    
    var start:Int64 = 1
    var end:Int64 = Int64(maxTime * n)
    var mid:Int64 = (start + end)/2
    
    while start < end {
        mid = (start + end)/2
        var capableN:Int64 = times.reduce(0) { $0 + mid/Int64($1) }
        if capableN >= n {
            end = mid
        } else {
            start = mid+1
        }
    }
    
    return end
}
